File: Mnema.Frontend/Web/src/app/_models/metadata-provider-settings.ts
import {MetadataBag} from "@mnema/_models/search";
import {MetadataProvider} from "@mnema/features/monitored-series/metadata.service";

export interface MetadataProviderSettings {
  id: string;
  metadataProvider: MetadataProvider;

  priority: number;

  enabled: boolean;

  seriesTitle: boolean;
  seriesSummary: boolean;
  seriesLocalizedName: boolean;
  seriesCoverUrl: boolean;
  seriesPublicationStatus: boolean;
  seriesAgeRating: boolean;
  seriesYear: boolean;
  seriesTags: boolean;
  seriesPeople: boolean;
  seriesLinks: boolean;

  chapters: boolean;
  chapterTitle: boolean;
  chapterSummary: boolean;
  chapterReleaseDate: boolean;
  chapterPeople: boolean;
  chapterTags: boolean;
  chapterCoverUrl: boolean;

  metadataProviderSpecific: MetadataBag;
}

File: Mnema.Models/Entities/MetadataProviderSettings.cs
using System;
using Microsoft.EntityFrameworkCore;
using Mnema.Common;
using Mnema.Models.Entities.Interfaces;
using Mnema.Models.Enums;

namespace Mnema.Models.Entities;

[Index(nameof(MetadataProvider), IsUnique = true)]
public class MetadataProviderSettings: IDatabaseEntity
{
    public Guid Id { get; set; }
    public MetadataProvider MetadataProvider { get; set; }

    public int Priority { get; set; }

    public bool Enabled { get; set; }

    public bool SeriesTitle { get; set; }

    public bool SeriesSummary { get; set; }

    public bool SeriesLocalizedName { get; set; }

    public bool SeriesCoverUrl { get; set; }

    public bool SeriesPublicationStatus { get; set; }

    public bool SeriesAgeRating { get; set; }

    public bool SeriesYear { get; set; }

    public bool SeriesTags { get; set; }

    public bool SeriesPeople { get; set; }

    public bool SeriesLinks { get; set; }

    public bool Chapters { get; set; }

    public bool ChapterTitle { get; set; }

    public bool ChapterSummary { get; set; }

    public bool ChapterReleaseDate { get; set; }

    public bool ChapterPeople { get; set; }

    public bool ChapterTags { get; set; }

    public bool ChapterCoverUrl { get; set; }

    public MetadataBag MetadataProviderSpecific { get; set; }

    public T GetKey<T>(IMetadataKey<T> key)
    {
        return MetadataProviderSpecific.GetKey(key);
    }
}
